### DISCUSSION TITLE
OpenVINO - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#4145] [Docs]
# OpenVINO - Ultralytics YOLOv8 Docs

Discover the power of deploying your Ultralytics YOLOv8 model using OpenVINO format for up to 10x speedup vs PyTorch.

https://docs.ultralytics.com/integrations/openvino/?h=openvino

**Comment by khanonenet** [comment]
How can we integrate this by converting the results into Detections using Supervision? I am sorry I am new to openVino but in the past I have used results by .pt models to get detections and perform object tracking within an ROI. Any help would be greatly appreciated.

**Reply by pderrenger** [reply]
@khanonenet integrating YOLOv8 with OpenVINO and converting the results into detections for use with Supervisely or similar platforms involves a few steps. Here's a concise guide:

1. **Export YOLOv8 to ONNX**: First, export your trained YOLOv8 model to ONNX format using the `export` mode with the `format='onnx'` argument.

    ```python
    from ultralytics import YOLO

    model = YOLO('path/to/your/model.pt')
    model.export(format='onnx')
    ```

2. **Convert ONNX to OpenVINO**: Use the OpenVINO Model Optimizer to convert the ONNX model to OpenVINO's Intermediate Representation (IR) format.

    ```bash
    mo --input_model path/to/your/model.onnx
    ```

3. **Inference with OpenVINO**: Load the IR model into OpenVINO's Inference Engine and perform inference to get the detections.

    ```python
    from openvino.inference_engine import IECore

    ie = IECore()
    net = ie.read_network(model='path/to/your/model.xml', weights='path/to/your/model.bin')
    exec_net = ie.load_network(network=net, device_name='CPU')  # or 'MYRIAD', 'GPU', etc.
    ```

4. **Process Detections**: After inference, process the output to extract the detection results, which typically include bounding boxes, confidence scores, and class IDs.

5. **Object Tracking**: To perform object tracking within an ROI, you can use the tracking algorithms provided by OpenVINO or integrate with a tracking library like OpenCV's tracking API or a custom tracker.

6. **Supervisely Integration**: To integrate with Supervisely, you'll need to format the detection results according to Supervisely's annotation format and then upload them to the platform.

Remember to refer to the Ultralytics Docs for detailed instructions on each step, and check out the Supervisely documentation for specifics on integrating detections into their platform. If you encounter any issues or have further questions, feel free to reach out! 😊🚀

**Comment by js333031** [comment]
How do I specify Arc or Flex GPU device when running `yolo benchmark model=yolov8n.pt data=coco128.yaml`? The device option expects CPU or a number for Nvidia GPU device.

**Reply by pderrenger** [reply]
@js333031 when running the YOLOv8 benchmark command, you can specify the device using the `device` argument. For NVIDIA GPUs, you would typically pass the GPU device index (e.g., `device=0` for the first GPU). However, for Arc or Flex GPUs, which are not directly supported by the benchmark command, you would still use the device index that corresponds to the GPU you want to use.

If you're running on a system where the Arc or Flex GPU is recognized as a CUDA-capable device by PyTorch, you can find the device index by running `torch.cuda.device_count()` and `torch.cuda.get_device_name(device)` in a Python shell to list all available CUDA devices and their names. Then, use the appropriate index in the benchmark command.

If the device is not recognized or you encounter any issues, please ensure that you have the correct drivers installed and that your system is configured properly to recognize the GPU. If problems persist, consider reaching out to the community or checking the documentation for any updates on compatibility and support for these specific GPUs. 🛠️

**Comment by Harly-1506** [comment]
Hello everyone,

Check out my new GitHub repository for running YOLOv8 object detection and segmentation inference using OpenVINO and NumPy only. This implementation is faster than the Torch version, offering improved performance and efficiency. Visit the repository here: [Faster Inference YOLOv8](https://github.com/Harly-1506/Faster-Inference-yolov8).

Your feedback and contributions are welcome!


Thanks

**Reply by pderrenger** [reply]
Hello,

Thanks for sharing your repository on YOLOv8 inference using OpenVINO and NumPy! It's great to see community-driven enhancements focusing on performance and efficiency. I'll definitely check it out and encourage others interested in optimized inference solutions to do the same. 🚀

Keep up the fantastic work, and I look forward to seeing how your project evolves with community feedback and contributions!

Best regards.
